tools/test/Makefile

```diff
@@ -41,8 +41,10 @@ tools/test/gen_print.c: $(TOOLS_WIRE_DEPS)
 	echo '#include "gen_test.h"' > $@
 	$(BOLT_GEN) -P --page impl ${@:.c=.h} test_type < tools/test/test_cases >> $@
 
-
+# Don't require update-mocks hack unless developer mode.
+ifeq ($(DEVELOPER),1)
```

Comment:
ideally this only gets built when make-check is called?

Reply:
Up until now we've build tests by default, so I didn't change this.